src/main/java/at/hannibal2/skyhanni/data/QuiverAPI.kt

```diff
@@ -0,0 +1,254 @@
+
+            val filledUpType = getByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return saveArrowAmount()
+        }
+
+        fillUpPattern.matchMatcher(message) {
+            val flintAmount = group("flintAmount").formatNumber().toFloat()
+            val existingAmount = arrowAmount[getByNameOrNull("ARROW".asInternalName())] ?: 0f
+            val newAmount = existingAmount + flintAmount
+
+            arrowAmount[getByNameOrFlint("ARROW".asInternalName())] = newAmount
+
+            return saveArrowAmount()
+        }
+
+        addedToQuiverPattern.matchMatcher(message) {
+            val type = group("type")
+            val amount = group("amount").formatNumber().toFloat()
+
+            val filledUpType = getByNameOrNull(type) ?: return
+
+            val existingAmount = arrowAmount[filledUpType] ?: 0f
+            val newAmount = existingAmount + amount
+            arrowAmount[filledUpType] = newAmount
+
+            return saveArrowAmount()
+        }
+
+        clearedPattern.matchMatcher(message) {
+            currentAmount = 0
+            arrowAmount.clear()
+
+            return saveArrowAmount()
+        }
+
+        arrowResetPattern.matchMatcher(message) {
+            currentArrow = getByNameOrNull("NONE".asInternalName()) ?: return
+            currentAmount = 0
+
+            saveArrowType()
+            return saveArrowAmount()
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryFullyLoaded(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (event.inventoryName != "Quiver") return
+
+        // clear to prevent duplicates
+        currentAmount = 0
+        arrowAmount.clear()
+
+        val stacks = event.inventoryItems
+        for (stack in stacks.values) {
+            val lore = stack.getLore()
+            if (lore.isEmpty()) continue
+
+            val arrow = stack.getInternalNameOrNull() ?: continue
+
+            val arrowType = getByNameOrNull(arrow) ?: continue
+            val arrowAmount = stack.stackSize + (this.arrowAmount[arrowType] ?: 0.0f)
+
+            this.arrowAmount[arrowType] = arrowAmount
+        }
+
+        saveArrowAmount()
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    // Inspired by SkyblockFeatures - https://github.com/MrFast-js/SkyblockFeatures/
+    fun onPlaySound(event: PlaySoundEvent) {
+        val holdingBow = InventoryUtils.getItemInHand()?.item is ItemBow && !fakeBows.contains(
+            InventoryUtils.getItemInHand()?.getInternalNameOrNull()
+        )
+
+        // check if sound location is more than configAmount block away from player
+        val soundLocation = event.location
+        if (soundLocation.distanceToPlayer() > SkyHanniMod.feature.dev.bowSoundDistance) return
```

Comment:
There is already a value for that
```suggestion
        val soundLocation = event.distanceToPlayer
        if (soundLocation > SkyHanniMod.feature.dev.bowSoundDistance) return
```